quickstarts/microsoft.healthcareapis/fhir-import/azuredeploy.json

```diff
@@ -0,0 +1,217 @@
+        }
+      },
+      "workspaceName": {
+        "type": "string",
+        "metadata": {
+          "description": "Workspace containing the Azure Health Data Services workspace"
+        }
+      },
+      "fhirName": {
+        "type": "string",
+        "metadata": {
+          "description": "Name of FHIR service"
+        }
+      },
+      "fhirKind": {
+        "type": "string",
+        "defaultValue": "fhir-R4",
+        "allowedValues": [
+          "fhir-R4",
+          "fhir-Stu3"
+        ],
+        "metadata": {
+          "description": "Kind of the FHIR service to update"
+        }
+      },
+      "storageName": {
+        "type": "string",
+        "defaultValue": "",
+        "metadata": {
+          "description": "Name of storage account to use for import. Needs to be an existing storage account. Leave blank if this has already been configured."
+        }
+      },
+      "enableImport": {
+        "type": "bool",
+        "metadata": {
+          "description": "Flag to enable or disable $import"
+        }
+      },
+      "newOrExisting": {
+        "type": "string",
+        "defaultValue": "existing",
+        "allowedValues": [
+            "new",
+            "existing"
+        ]
+      }
+    },
+    "variables": {
+      "enableConfiguration": {
+        "enabled": true,
+        "initialImportMode": true,
+        "integrationDataStore": "[parameters('storageName')]"
+      },
+      "disableConfiguration": {
+        "enabled": false,
+        "initialImportMode": false
+      },
+      "newdeployname": "[format('newdeploy{0}', uniqueString(resourceGroup().id, parameters('fhirName')))]",
+      "existingdeployname": "[format('existingdeploy{0}', uniqueString(resourceGroup().id, parameters('fhirName')))]"
+    },
+    "resources": [
+      {
+        "type": "Microsoft.HealthcareApis/workspaces/fhirservices",
+        "apiVersion": "2022-06-01",
+        "name": "[format('{0}/{1}', parameters('workspaceName'), parameters('fhirName'))]",
+        "location": "[parameters('resourceLocation')]",
+        "kind": "[parameters('fhirKind')]",
+        "identity": {
+          "type": "SystemAssigned"
+        },
+        "properties": "[union(reference(resourceId('Microsoft.Resources/deployments', if(equals(parameters('newOrExisting'), 'existing'), variables('existingdeployname'), variables('newdeployname'))), '2020-10-01').outputs.properties.value, createObject('importConfiguration', if(parameters('enableImport'), variables('enableConfiguration'), variables('disableConfiguration'))))]",
+        "dependsOn": [
+          "[resourceId('Microsoft.Resources/deployments', if(equals(parameters('newOrExisting'), 'existing'), variables('existingdeployname'), variables('newdeployname')))]"
+        ],
+        "metadata": {
+          "description": "Updated FHIR Service used to enable import"
+        }
+      },
+      {
+        "condition": "[equals(parameters('newOrExisting'), 'existing')]",
```

Comment:
what problem are we trying to solve with the exclusive deployments?  ARM is declarative, so new/existing should not be different code (unless the resource provider is not behaving correctly)?

Reply:
For this change, we target to provide the change toggle for the feature, so we prefer to keep existing properties and the "new" is not recommended, and would like to create a new resource of default settings for customer.

Reply:
@bmoore-msft Above is the answer to your comment. Could you review again and approve?

Reply:
There should be only one declaration for what you're trying to do - if the resource is not modeled correct such that you can simply add the property, then make the following changes:

- move the creation of the workspace to the main template - it does not need to be nested - this will still have the condition on it
- leave the nested deployment for getting the existing settings
- the `fhirservices` resource will need to dependOn both resources (no coditions in the dependsOn)
- the properties property of the `fhirservices` resource should start with the `if()` statement